datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -147,6 +149,24 @@ impl PhysicalGroupBy {
     }
 }
 
+impl PartialEq for PhysicalGroupBy {
+    fn eq(&self, other: &PhysicalGroupBy) -> bool {
+        self.expr.len() == other.expr.len()
+            && self
+                .expr
+                .iter()
+                .zip(other.expr.iter())
+                .all(|((expr1, name1), (expr2, name2))| expr1.eq(expr2) && name1 == name2)
```

Comment:
I wondered why this needed to be manually derived, so I tried removing it and got this error:

```
error[E0369]: binary operation `==` cannot be applied to type `Vec<(Arc<dyn PhysicalExpr>, std::string::String)>`
  --> datafusion/core/src/physical_plan/aggregates/mod.rs:91:5
   |
88 | #[derive(Clone, Debug, Default, PartialEq)]
   |                                 --------- in this derive macro expansion
...
91 |     expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)

```

Reply:
It looks like if Struct contains any dynamic trait object, we can not use the `PartialEq`  derive macros.

https://github.com/rust-lang/rust/issues/39128